### PR TITLE
feat: add windows amd64 installer

### PR DESCRIPTION
**Pull request by deansheather**
Adds a Windows amd64 installer (built on Linux) using `makensis`. NSIS is an alternative to MSI which can easily be built on platforms other than Windows, and is used by well-known apps such as VLC and VirtualBox.

Drops the following files in `C:\Program Files\Coder`:
- `bin/coder.exe`
- `uninstaller.exe`
- `license.txt`

Optionally adds `bin` to PATH. Registers Coder as an app in Windows settings so users can easily uninstall it.

https://user-images.githubusercontent.com/11241812/197573088-2db4051c-fd32-4092-b16b-923f859e3be6.mp4

### CODE REVIEW

**Comment by kylecarbs** [comment]
Does the user need to sign out and sign in for the `PATH` to change?

File: scripts/installer/path.nsh [hunk 1]
```diff
@@ -0,0 +1,190 @@
+# PATH utilities. Taken from:
```
Comment (kylecarbs) [inline scripts/installer/path.nsh:1]: Should we change this dir to `win-installer` or even just `windows` instead?

**Comment by deansheather** [comment]
> Does the user need to sign out and sign in for the PATH to change?

@kylecarbs It's the easiest way to get PATH updates to apply without having to run crazy commands to force refresh the PATH variable AFAIK

**Review by kylecarbs — APPROVED** [review]
This is awesome!

Could you adjust this in the README? https://github.com/coder/coder#getting-started

Add under:

```
The easiest way to install Coder is to use our [install script](https://github.com/coder/coder/blob/main/install.sh) for Linux and macOS. For Windows, use our [installer](<link to latest release installer bin here>).
```

File: scripts/installer/path.nsh (hunk 1 above)
Comment (deansheather) [reply scripts/installer/path.nsh]: Renamed to `win-installer`.

**Comment by deansheather** [comment]
@kylecarbs there's no permalink for the installer, so I put the following in the README instead:

```
For Windows, use the latest `..._installer.exe` file from GitHub Releases.
```

**Comment by kylecarbs** [comment]
@deansheather hmm, that makes sense to me. link to the latest release too!

Thoughts on this? https://stackoverflow.com/questions/6979741/setting-environment-variables-requires-reboot-on-64-bit

btw this isn't blocking, just cool if it's easy to do

**Comment by kylecarbs** [comment]
@deansheather can we include the version somewhere in the installer UI? Might be helpful for folks

**Comment by matifali** [comment]
Also, it will alloww publishing on `winget` as currently, it does not allow publishing a portable app from inside a zip.
#3729

**Comment by deansheather** [comment]
yeo @matifali this is a follow up I'd like to do after merge :D

**Review by zounce — APPROVED** [review]

**Pull request merged**